clients/cfrestclient/rest_cloud_foundry_client_extended.go

```diff
@@ -68,23 +157,136 @@ func (c CloudFoundryRestClient) getSharedDomainsInternal(cloudFoundryUrlElements
 	}
 	nextUrlQuery := nextUrl.Query()
 	page := nextUrlQuery.Get("page")
-	resultsPerPage := nextUrlQuery.Get("results-per-page")
-	orderDirection := nextUrlQuery.Get("order-direction")
+	resultsPerPage := nextUrlQuery.Get("per_page")
 
-	return CloudFoundryUrlElements{Page: &page, ResultsPerPage: &resultsPerPage, OrderDirection: &orderDirection}, nil
+	return CloudFoundryUrlElements{Page: &page, ResultsPerPage: &resultsPerPage}, nil
 
 }
 
-func toSharedDomains(response *models.CloudFoundryResponse) []models.SharedDomain {
-	var result []models.SharedDomain
+func toSharedDomains(response *models.CloudFoundryResponse) []models.Domain {
+	var result []models.Domain
 	for _, cloudResource := range response.Resources {
-		result = append(result, models.NewSharedDomain(cloudResource.Entity.Name, cloudResource.Metadata.GUID, cloudResource.Metadata.URL))
+		result = append(result, models.NewDomain(cloudResource.Name, cloudResource.GUID))
 	}
 	return result
 }
+
+func getPaginatedResources[T any](url, token string) ([]T, error) {
+	var result []T
+	for url != "" {
+		body, err := sendRequestAndReadBody(url, token)
+		if err != nil {
+			return nil, err
+		}
+		response, err := parseBody[models.PaginatedResponse[T]](body)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, entity := range response.Resources {
+			result = append(result, entity)
+		}
+		url = response.Pagination.NextPage
+	}
+	return result, nil
+}
+
+func getPaginatedResourcesWithIncluded[T any, Auxiliary any](url, token string, auxiliaryContentHandler func(T, Auxiliary) T) ([]T, error) {
+	var result []T
+	for url != "" {
+		body, err := sendRequestAndReadBody(url, token)
+		if err != nil {
+			return nil, err
+		}
+		response, err := parseBody[models.PaginatedResponseWithIncluded[T, Auxiliary]](body)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, entity := range response.Resources {
+			result = append(result, auxiliaryContentHandler(entity, response.Included))
+		}
+		url = response.Pagination.NextPage
+	}
+	return result, nil
+}
+
+func sendRequestAndReadBody(url, token string) ([]byte, error) {
+	req, _ := http.NewRequest(http.MethodGet, url, nil)
+	req.Header.Add("Authorization", token)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil { // this will be set if the client failed to send the request or receive the response
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	bytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode/100 != 2 {
+		return nil, models.HttpResponseError{Underlying: fmt.Errorf("%s: %s", resp.Status, string(bytes))}
+	}
+	return bytes, nil
+}
+
+func parseBody[T any](body []byte) (T, error) {
+	var result T
+	err := jsonry.Unmarshal(body, &result)
+	if err == nil {
+		return result, nil
+	}
+	//jsonry doesn't work with raw objects like map, so try the base json decoder
```

Comment:
Why are we even using it then? Can't we just use the base json parser

Reply:
Because it is useful :D 
De/Serializing relationships is a pain manually - e.g.:
```golang
type App struct {
    ...
    Relationships appRel `json:"relationships"`
}

type appRel struct {
    Space toOneRel `json:"space"`
}

type toOneRel struct {
    Data data `json:"data"`
}

type data struct {
    Guid string `json:"guid"`
}
```

Whereas with _jsonry_ we can just do:
```golang
type App struct {
    ...
    SpaceGuid string `jsonry:"relationships.space.data.guid"`
}
```